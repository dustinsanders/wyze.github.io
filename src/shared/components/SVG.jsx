// @flow

import type { VNode } from '../../types'
import preact from 'preact'

type Props = {
  children?: VNode,
  height: number,
  width: number,
}

const SVG = ({ children, height, width }: Props): VNode => (
  <svg
    height="100%"
    viewBox={`0 0 ${width} ${height}`}
    xmlns="http://www.w3.org/2000/svg"
  >
    {children}
  </svg>
)

SVG.defaultProps = {
  children: null,
}

export default SVG
