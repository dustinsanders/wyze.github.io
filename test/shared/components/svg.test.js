// @flow

import SVG from '../../../src/shared/components/SVG'
import preact from 'preact'
import snapshot from '../../helpers/snapshot'

const svg = {
  // eslint-disable-next-line react/jsx-filename-extension
  children: <g />,
  height: 100,
  width: 100,
}

describe('<SVG />', () => {
  it('renders', snapshot.bind(null, SVG, svg))
})
