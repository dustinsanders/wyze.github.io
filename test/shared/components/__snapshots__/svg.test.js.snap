// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SVG /> renders 1`] = `
"<svg height=\\"100%\\" viewBox=\\"0 0 100 100\\" xmlns=\\"http://www.w3.org/2000/svg\\">
  <g></g>
</svg>"
`;
